app/frontend/src/components/Comments/CommentBox.tsx

```diff
@@ -0,0 +1,118 @@
+
+const useStyles = makeStyles((theme) => ({
+  card: {
+    marginTop: "1em",
+    padding: "1em",
+    border: "1px solid",
+  },
+}));
+
+interface CommentBoxProps {
+  threadId: number;
+}
+
+// Reply with more Reply objects as children
+interface MultiLevelReply extends Reply.AsObject {
+  children: Array<Reply.AsObject>;
+  // page token, etc? not sure what's needed for react query
+}
+
+export default function CommentBox({ threadId }: CommentBoxProps) {
+  const classes = useStyles();
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const [comments, setComments] = useState<Array<MultiLevelReply>>([]);
+
+  useEffect(() => {
+    (async () => {
+      setLoading(true);
+      try {
+        const thread = await service.threads.getThread(threadId);
+        setComments(
+          await Promise.all(
+            thread.repliesList.map(async (reply) => {
+              return {
+                ...reply,
+                children:
+                  reply.numReplies > 0
+                    ? (await service.threads.getThread(reply.threadId))
+                        .repliesList
+                    : [],
+              };
+            })
+          )
+        );
+      } catch (e) {
+        console.error(e);
+        setError(e.message);
+      }
+      setLoading(false);
+    })();
+  }, [threadId]);
+
+  const onComment = async (threadId: number, content: string) => {
+    await service.threads.postReply(threadId, content);
+    setLoading(true);
+    try {
+      const thread = await service.threads.getThread(threadId);
+      setComments(
+        await Promise.all(
+          thread.repliesList.map(async (reply) => {
+            return {
+              ...reply,
+              children:
+                reply.numReplies > 0
+                  ? (await service.threads.getThread(reply.threadId))
+                      .repliesList
+                  : [],
+            };
+          })
+        )
+      );
+    } catch (e) {
+      console.error(e);
+      setError(e.message);
+    }
+    setLoading(false);
+  };
+  return (
+    <>
+      {error && <Alert severity="error">{error}</Alert>}
+      {loading && <CircularProgress />}
+      {comments.map((comment) => (
+        <>
+          <Card className={classes.card}>
+            Comment: by user id {comment.authorUserId}, posted at{" "}
+            {comment.createdTime!.seconds}, {comment.numReplies} replies.
+            <Markdown source={comment.content} />
+            Replies:
+            {comment.children.map((reply) => (
+              <>
+                <Card className={classes.card}>
+                  Reply: by user id {reply.authorUserId}, posted at{" "}
+                  {reply.createdTime!.seconds}.
+                  <Markdown source={reply.content} />
+                </Card>
+              </>
+            ))}
+            <NewComment
+              onComment={(content) => onComment(comment.threadId, content)}
```

Comment:
```suggestion
              onComment={(content) => handleComment(comment.threadId, content)}
```
Convention is the event is `onEvent` and the function is `handleEvent` (though are other places in the code which don't currently reflect this)